DetailedMapTacks.modinfo

```diff
@@ -71,7 +68,7 @@
     <File>text/dmt_translations_text.xml</File>
     <!-- Game files -->
     <File>config/dmt_config.xml</File>
-    <File>gameplay/mappinsubjectmanager.lua</File>
+    <File>ui/dmt_mappinsubjectmanager.lua</File>
```

Comment:
Also wonder how do we consider backward compatibility, i.e. make sure players who has a in-progress game, they won't find the map pin yields missing after updating the mod. 

If we do find serialization performance is a concern, we can always have single player continue to use the existing player property solution, and only apply the new logic if it is a multiplayer game.

Reply:
Thankfully, backwards compatibility isn't an issue with these changes. Existing games will still have the pin data saved in the player properties, but just won't be used anymore. Instead on the first load for existing saves, it'll check the player configuration for the pin data, find that there's nothing there, and then it'll regenerate all of the data. This is handled in the `OnLocalPlayerTurnBegin` function in `dmt_yieldcalculater.lua`. That same code made to add support for saves made before adding the mod works for this purpose too.

The only concern I could see is that any data saved in the properties from before is just wasted memory now. Though it's pretty insignificant relative to the memory usage of the game itself. Could add a new function to clear it out if it's an issue though.

As for the serialization performance, singleplayer was never an issue. It was only multiplayer that seemed to have a slight regression in performance. It seems to be something behind the scenes, so I can't tell exactly why though. In any case, the new serialization method I implemented based on your suggestion works even better than the previous one. With all that, I don't think it's necessary to apply separate logic for singleplayer vs multiplayer.